pkg/daemon/update.go

```diff
@@ -792,6 +807,94 @@ func (dn *Daemon) switchKernel(oldConfig, newConfig *mcfgv1.MachineConfig) error
 	return nil
 }
 
+func addExtRepo(mnt string) error {
+	if err := os.MkdirAll(filepath.Dir(extensionsRepo), 0755); err != nil {
+		return fmt.Errorf("Error creating dir %s: %v", filepath.Dir(extensionsRepo), err)
+	}
+	repo, err := os.Create(extensionsRepo)
+	if err != nil {
+		repo.Close()
+		return err
+	}
+	defer repo.Close()
+	if _, err := repo.WriteString("[rhcos-extensions]\nenabled=1\nmetadata_expire=1m\nbaseurl=" + mnt + "/extensions/\ngpgcheck=0\nskip_if_unavailable=False\n"); err != nil {
+		return err
+	}
+	return nil
+}
+
+func generateExtensionsArgs(oldConfig, newConfig *mcfgv1.MachineConfig) []string {
```

Comment:
would like a comment here just so it's obvious what this is

Reply:
Thanks for the suggestion!
Until this is feature complete, there is possibility of changes in the logic if needed. Will be adding comments at needed places once we cover all scenarios.

Reply:
sounds perf!